castai/utils.go

```diff
@@ -120,3 +125,20 @@ func normalizeJSON(bytes []byte) ([]byte, error) {
 	}
 	return json.Marshal(output)
 }
+
+func getDefaultOrganizationId(ctx context.Context, meta any) (string, error) {
```

Comment:
Unfortunately yes. In the current `castai_reservations` resource organization ID is passed as an optional argument because the endpoints consume it (it uses InventoryAPI):
![image](https://github.com/castai/terraform-provider-castai/assets/5285849/952d8693-0a87-4a65-aa19-f31bdec4f9d0)

In the CommitmentsAPI though, we no longer have the argument meaning that the default organization ID associated with the auth token is used. So if a user has access to multiple organizations, the flow is to use multiple `castai` providers with different auth tokens.